hazelcast-client/src/test/java/com/hazelcast/client/ClientTimeoutTest.java

```diff
@@ -35,24 +36,32 @@
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
 
-@RunWith(HazelcastParallelClassRunner.class)
+/**
+ * Tests in this class intentionally use real network.
+ */
+@RunWith(HazelcastSerialClassRunner.class)
 @Category({QuickTest.class})
 public class ClientTimeoutTest {
 
-    private final TestHazelcastFactory hazelcastFactory = new TestHazelcastFactory();
-
     @After
     public void cleanup() {
-        hazelcastFactory.terminateAll();
+        Hazelcast.shutdownAll();
+        HazelcastClient.shutdownAll();
     }
 
-
     @Test(timeout = 20000, expected = IllegalStateException.class)
-    public void testTimeoutToOutsideNetwork() throws Exception {
+    public void testTimeoutToOutsideNetwork() {
         ClientConfig clientConfig = new ClientConfig();
         clientConfig.getGroupConfig().setName("dev").setPassword("dev-pass");
-        clientConfig.getNetworkConfig().addAddress("8.8.8.8:5701");
-        HazelcastInstance client = hazelcastFactory.newHazelcastClient(clientConfig);
+
+        ClientNetworkConfig networkConfig = clientConfig.getNetworkConfig();
+        networkConfig.addAddress("8.8.8.8:5701");
+        // Do only one connection-attempt
+        networkConfig.setConnectionAttemptLimit(1);
+        // Timeout connection-attempt after 1000 millis
+        networkConfig.setConnectionTimeout(1000);
```

Comment:
Same test for C++ client also sets the connectıonAttemptPeriod: https://github.com/hazelcast/hazelcast-cpp-client/blob/master/hazelcast/test/src/cluster/ClientConnectionTest.cpp#L53

You may also set it to speed up the test. Default is 3 seconds.